cilium/bpf_metadata.cc

```diff
@@ -349,7 +358,8 @@ const PolicyInstanceConstSharedPtr Config::getPolicy(const std::string& pod_ip)
   return policy;
 }
 
-Cilium::SocketOptionSharedPtr Config::getMetadata(Network::ConnectionSocket& socket) {
+Cilium::BpfMetadata::SocketInformationSharedPtr
+Config::extractSocketInformation(Network::ConnectionSocket& socket) {
```

Comment:
Currently, extraction of the socket information is more or left as is. Just being renamed and returning an "intermediate" struct holding the required data that is then used to create the actual socket options.

It also still performs some actions on the socket itself.

* restoration of the local address (for the original dst cluster)
* setting the requested application protocol that is configured with proxylib

IMO extracting information shouldn't perform any actions on the socket at all. Probably worth to clean this up too (and move outside of this function) - maybe in a follow up PR.

Reply:
I like the term `metadata` vs. `information`, as it is derived from Cilium bpf datapath resources associated with the socket, rather than from the socket itself! So if you want to rename this, how about `extractSocketMetadata`?

Reply:
makes sense. will rename the function & intermediate struct accordingly

Reply:
renamed